src/Arkanis.Overlay.Host.Desktop/Workers/GameWindowTracker.cs

```diff
@@ -125,6 +69,50 @@ ILogger<WindowTracker> logger
 
     private uint _threadId;
 
+    public GameWindowTracker(
+        IUserPreferencesProvider userPreferencesProvider,
+        IHostApplicationLifetime applicationLifetime,
+        ILogger<GameWindowTracker> logger
+    )
+    {
+        _applicationLifetime = applicationLifetime;
+        _userPreferencesProvider = userPreferencesProvider;
+        _windowSizeAndPositionDebounceTimer = new Timer(OnDebounceTimer_UpdateWindowSizeAndPosition);
+        _logger = logger;
+
+        ProcessExited += OnProcessExited;
+        WindowFound += OnWindowFound;
+        WindowLost += OnWindowLost;
+    }
+
+    public Size CurrentWindowSize { get; private set; }
+    public Point CurrentWindowPosition { get; private set; }
+    public bool IsWindowFocused { get; private set; }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+    private HWND CurrentWindowHWnd
+    {
+        get => _currentWindowHWnd;
+        set
+        {
+            _currentWindowHWnd = value;
+            DispatchFast(() =>
+                {
+                    if (value != HWND.Null)
+                    {
+                        // an active window is tracked
+                        Invoke(() => WindowFound?.Invoke(this, value));
+                    }
+                    else
+                    {
+                        // the tracked window was lost
+                        Invoke(() => WindowLost?.Invoke(this, EventArgs.Empty));
```

Comment:
Invoke is not needed here, since this should never be invoked from a separate thread.